### PR TITLE
DEVEXP-489: Rebase k8s 117

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @adambkaplan 

@openshift/openshift-team-developer-experience fyi

### CODE REVIEW

**Comment by gabemontero** [comment]
@soltysh @mfojtik please let me know if you want someone from your respective teams to be officially assigned to this PR

File: pkg/cmd/openshift-controller-manager/controller_manager.go [hunk 1]
```diff
@@ -25,9 +25,6 @@ import (
 	origincontrollers "github.com/openshift/openshift-controller-manager/pkg/cmd/controller"
 	"github.com/openshift/openshift-controller-manager/pkg/cmd/imageformat"
 	"github.com/openshift/openshift-controller-manager/pkg/version"
-
-	// for metrics
-	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus"
```
Comment (adambkaplan) [inline pkg/cmd/openshift-controller-manager/controller_manager.go]: @deads2k was this import used for anything? If so, is there something to replace it?

File: pkg/cmd/controller/apiserver_handlers.go [hunk 2]
```diff
@@ -16,9 +18,9 @@ func newBypassAuthorizer(auth kauthorizer.Authorizer, paths ...string) kauthoriz
 	return bypassAuthorizer{paths: sets.NewString(paths...), authorizer: auth}
 }
 
-func (a bypassAuthorizer) Authorize(attributes kauthorizer.Attributes) (allowed kauthorizer.Decision, reason string, err error) {
+func (a bypassAuthorizer) Authorize(ctx context.Context, attributes kauthorizer.Attributes) (allowed kauthorizer.Decision, reason string, err error) {
```
Comment (adambkaplan) [inline pkg/cmd/controller/apiserver_handlers.go]: /cc @deads2k

File: pkg/cmd/openshift-controller-manager/controller_manager.go (hunk 1 above)
Comment (gabemontero) [reply pkg/cmd/openshift-controller-manager/controller_manager.go]: Did a little more research on this one @adambkaplan 

that `_` import was done very early on in the ocm historywith https://github.com/openshift/openshift-controller-manager/commit/b7179b40d0ecbe386d9a8a079fc2224a5505cf62 as part of the `v0.0.0-alpha0` tag back in Nov 2018.

Today, there are absolutely no other references to that package in the repo.

And across the board now all the controllers in OCM are using `github.com/prometheus/client_golang/prometheus`

Assuming @deads2k is spending his cycles elsewhere I think we can assume this change is in the benign / cleanup category, especially given that it no longer exists in k8s/k8s.

File: pkg/cmd/controller/apiserver_handlers.go (hunk 2 above)
Comment (gabemontero) [reply pkg/cmd/controller/apiserver_handlers.go]: fyi @adambkaplan this change was needed as a result of an upstream change in k8s apiserver

see https://github.com/kubernetes/apiserver/blob/kubernetes-1.17.0-rc.1/pkg/authorization/authorizer/interfaces.go#L71

File: pkg/cmd/openshift-controller-manager/controller_manager.go (hunk 1 above)
Comment (adambkaplan) [reply pkg/cmd/openshift-controller-manager/controller_manager.go]: If it's no longer upstream, then I am going to assume this is a no-op. Good news is that we do have test coverage on the build metrics in our conformance suite, and this is green.

**Pull request merged**